src/message/messagesReducer.js

```diff
@@ -83,9 +83,9 @@ export default (state: MessagesState = initialState, action: Action): MessagesSt
     case EVENT_REACTION_ADD:
       return state.update(
         action.message_id,
-        oldMessage =>
+        <M: Message>(oldMessage: M): M =>
           oldMessage && {
-            ...oldMessage,
+            ...(oldMessage: M),
```

Comment:
Similarly, this cast seems not to be needed.

Reply:
For this one (and likely the others in this file), I find that it's not needed in the commit that introduces them, but it *is* needed as of the commit that splits `type` between `PmMessage` and `StreamMessage`.